nestkernel/node.cpp

```diff
@@ -69,30 +69,26 @@ Node::~Node()
 {
 }
 
-void
-Node::init_state()
-{
-  Model const* const model = kernel().model_manager.get_model( model_id_ );
-  assert( model );
-  init_state_( model->get_prototype() );
-}
-
 void
 Node::init_state_( Node const& )
 {
 }
 
 void
-Node::init_buffers()
+Node::init()
 {
-  if ( buffers_initialized_ )
+  if ( state_buffers_initialized_ )
   {
     return;
   }
 
+  Model const* const model = kernel().model_manager.get_model( model_id_ );
+  assert( model );
+  init_state_( model->get_prototype() );
```

Comment:
The only reason to pass the prototype was to be able to copy values from it as part of `ResetNetwork`.  This is no longer relevant and `init_state_()` will only be called on nodes that have not been "touched" since creation, except possibly for SetStatus calls. Therefore, the node has all information internally and we should drop the argument to `init_state_()` entirely.